sdk/protocol2.go

```diff
@@ -0,0 +1,173 @@
+package sdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/cache"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/infra-integrations-sdk/metric"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// EntityData defines all the data related to a particular event from an entity.
+type EntityData struct {
+	Entity    Entity             `json:"entity"`
+	Metrics   []metric.MetricSet `json:"metrics"`
```

Comment:
in protocol 1 we have 
`Metrics            []*metric.MetricSet `json:"metrics"`
Why in one place we use a pointer and in the other we don't?

Reply:
The pointer is not needed since we are dealing with slices.

See https://github.com/newrelic/infra-integrations-beta/blob/master/integrations/kube-state-metrics/src/kube-state-metrics.go#L64